packages/rum-recorder/src/domain/deflateSegmentWriter.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { noop } from '@datadog/browser-core'
+
+import { SegmentMeta } from '../types'
+import { DeflateSegmentWriter } from './deflateSegmentWriter'
+import { DeflateWorker, DeflateWorkerAction, DeflateWorkerListener } from './deflateWorker'
+
+describe('DeflateWriter', () => {
+  let worker: MockWorker
+
+  beforeEach(() => {
+    worker = new MockWorker()
+  })
+
+  it('calls the onWrote callback when data is written', () => {
+    const onWroteSpy = jasmine.createSpy<(size: number) => void>()
+    const writer = new DeflateSegmentWriter(worker, onWroteSpy, noop)
+    writer.write('foo')
+    worker.process()
+    expect(onWroteSpy.calls.allArgs()).toEqual([[3]])
+  })
+
+  it('calls the onCompleted callback when data is complete', () => {
+    const onCompletedSpy = jasmine.createSpy<(data: Uint8Array, meta: SegmentMeta) => void>()
+    const writer = new DeflateSegmentWriter(worker, noop, onCompletedSpy)
+    const meta: SegmentMeta = { start: 12 } as any
+    writer.complete(undefined, meta)
+    worker.process()
+    expect(onCompletedSpy.calls.allArgs()).toEqual([[jasmine.any(Uint8Array), meta]])
+  })
+
+  it('calls the onCompleted callback with the correct meta even if a previous action failed somehow', () => {
```

Comment:
No, this is here to prevent very hard to understand issues if something goes wrong in the worker. It should not happen, but this way we have a guarantee that correct meta data is associated with the segment.

Reply:
I think we could be more explicit on what is the sanity check we want to perform and the recover strategy if it fails.
We should also probably send an alert to monitor this situation.

Reply:
FTR: add doc in Writer, add monitoring log in case of communication error, open future task for monitoring worker